src/java.desktop/macosx/classes/sun/lwawt/macosx/LWCToolkit.java

```diff
@@ -503,7 +502,7 @@ public Insets getScreenInsets(final GraphicsConfiguration gc) {
     @Override
     public void sync() {
         // flush the OGL pipeline (this is a no-op if OGL is not enabled)
-        if (MacOSFlags.isMetalEnabled()) {
+        if (CGraphicsDevice.useMetalPipeline()) {
             MTLRenderQueue.sync();
```

Comment:
Does this check mean that the MTLRenderQueue.sync() is not a noop if the metal is not enabled?

Reply:
> Does this check mean that the MTLRenderQueue.sync() is not a noop if the metal is not enabled?

I don't know how else we will differentiate between Metal and OGL here.
MTLRenderQueue.sync() is still a noop - as there is null instance check inside it.
The comment above the if check has become stale - I will update it.